### PR TITLE
refactor: remove connect-go from flagd-proxy and replace with grpc

### PR DESCRIPTION
**Pull request by james-milligan**
## This PR

- during load testing of the flags-proxy it was found that connect-go causes `connection reset by peer` errors when there are too many simultaneous connections, this is not the case when only using grpc. This PR removes connect-go, replacing it with a standalone grpc server
- the `fetchAllFlags` endpoint will no longer be available via a http request, this was only used internally by the flagd grpc sync implementation

### CODE REVIEW

File: core/pkg/service/sync/handler.go [hunk 1]
```diff
@@ -3,40 +3,42 @@ package sync
...
 
 type handler struct {
+	rpc.UnimplementedFlagSyncServiceServer
 	syncStore *syncStore.SyncStore
 	logger    *logger.Logger
 }
 
-func (l *handler) FetchAllFlags(ctx context.Context, req *connect.Request[syncv1.FetchAllFlagsRequest]) (
-	*connect.Response[syncv1.FetchAllFlagsResponse],
+func (l handler) FetchAllFlags(ctx context.Context, req *syncv1.FetchAllFlagsRequest) (
```
Comment (Kavindu-Dodan) [inline core/pkg/service/sync/handler.go:19]: Make a pointer receiver

**Comment by Kavindu-Dodan** [comment]
Overall like the idea of moving to pure grpc

File: core/pkg/service/sync/handler.go [hunk 2]
```diff
@@ -3,40 +3,42 @@ package sync
...
+	rpc "buf.build/gen/go/open-feature/flagd/grpc/go/sync/v1/syncv1grpc"
 	syncv1 "buf.build/gen/go/open-feature/flagd/protocolbuffers/go/sync/v1"
-	"github.com/bufbuild/connect-go"
 	"github.com/open-feature/flagd/core/pkg/logger"
 	"github.com/open-feature/flagd/core/pkg/sync"
 	syncStore "github.com/open-feature/flagd/core/pkg/sync-store"
 )
 
 type handler struct {
+	rpc.UnimplementedFlagSyncServiceServer
```
[the remark this answers is not in the document]
Comment (Kavindu-Dodan) [reply core/pkg/service/sync/handler.go:14]: I was missing the reason and it is for forward compatibility. Came across this [1] if anyone is interested. 

[1] - https://github.com/grpc/grpc-go/issues/3669

**Review by Kavindu-Dodan — APPROVED** [review]

**Review by toddbaert — APPROVED** [review]

**Comment by beeme1mr** [comment]
Does this only affect the proxy?

**Comment by toddbaert** [comment]
@beeme1mr 

> Does this only affect the proxy?

Technically? No. The [fetchAllFlags](https://github.com/open-feature/schemas/blob/main/protobuf/sync/v1/sync_service.proto#L58-L77) has been removed from the connect API and just uses plain gRPC. Practically though this doesn't impact flagd providers, just communication between flagd and a gRPC sync provider.

TBH I think this is a better choice... having the gRPC sync be implemented entirely as raw gRPC seems better... we know it will never be used in a web context, etc.

**Comment by james-milligan** [comment]
> @beeme1mr
> 
> > Does this only affect the proxy?
> 
> Technically? No. The [fetchAllFlags](https://github.com/open-feature/schemas/blob/main/protobuf/sync/v1/sync_service.proto#L58-L77) has been removed from the connect API and just uses plain gRPC. Practically though this doesn't impact flagd providers, just communication between flagd and a gRPC sync provider.
> 
> TBH I think this is a better choice... having the gRPC sync be implemented entirely as raw gRPC seems better... we know it will never be used in a web context, etc.
> 
> @james-milligan correct me if I'm wrong.

This is true, no SDKs will be routing directly to a proxy pod, but would rather be connecting to a standalone flagd pod using a grpc sync with a separate proxy instance, although the duplicated FetchAllFlags endpoint (exists in both the proxy and flagd) may be have the potential to cause confusion?

**Pull request merged**